apps/bfd-server/bfd-server-docs/src/main/java/gov/cms/bfd/datadictionary/CsvToExcelCommand.java

```diff
@@ -0,0 +1,326 @@
+        if (row.getRowNum() == 0) {
+          max = 2;
+        }
+        row.setHeight((short) (max * defaultHeight));
+      }
+    }
+  }
+
+  /**
+   * Determines the maximum number of lines in a row.
+   *
+   * @param row the row to inspect
+   * @param lastCol the last column in the row to inspect
+   * @return the maximum number of lines for any cell in a row (min:1)
+   */
+  private static short maxLinesInRow(Row row, int lastCol) {
+    short maxLines = 1;
+    for (int i = 0; i <= lastCol; i++) {
+      var cell = row.getCell(i);
+      if (cell != null) {
+        var lines = cell.getStringCellValue().split("\r\n|\n|\r").length;
+        if (maxLines < lines) {
+          maxLines = (short) lines;
+        }
+      }
+    }
+    return maxLines;
+  }
+
+  /**
+   * Do a text search and replace on a range of cells.
+   *
+   * @param sheet the sheet to search/replace
+   * @param range the range of cells to search/replace
+   * @param search the text to search for in each cell
+   * @param replace the text to replace the searched values with
+   */
+  private static void searchAndReplace(
+      Sheet sheet, CellRangeAddress range, String search, String replace) {
+    applyToRange(
+        sheet,
+        range,
+        (c) -> {
+          c.setCellValue(c.getStringCellValue().replace(search, replace));
+        });
+  }
+
+  /**
+   * Set the style for a range of cells.
+   *
+   * @param sheet the sheet to format
+   * @param range the range of cells within the sheet to format
+   * @param style the CellStyle to apply to the cells
+   */
+  private static void setStyle(Sheet sheet, CellRangeAddress range, CellStyle style) {
+    applyToRange(
+        sheet,
+        range,
+        (c) -> {
+          c.setCellStyle(style);
+        });
+  }
+
+  /**
+   * Implement alternate row shading.
+   *
+   * @param sheet the sheet to format
+   * @param range the range of cells with the sheet to format
+   * @param style the CellStyle to apply to alternate rows (odd numbered)
+   */
+  private static void rowShading(Sheet sheet, CellRangeAddress range, CellStyle style) {
+    applyToRange(
+        sheet,
+        range,
+        (c) -> {
+          if (c.getRowIndex() % 2 == 1) {
+            c.setCellStyle(style);
+          }
+        });
+  }
+
+  /**
+   * Apply a consumer function to a range of cells in a sheet.
+   *
+   * @param sheet the sheet to update
+   * @param range the range of cells to update
+   * @param fn a consumer function that takes a cell and returns void
+   */
+  private static void applyToRange(Sheet sheet, CellRangeAddress range, Consumer<Cell> fn) {
+    for (int i = range.getFirstRow(); i <= range.getLastRow(); i++) {
+      var row = sheet.getRow(i);
+      if (row != null) {
+        for (int j = range.getFirstColumn(); j <= range.getLastColumn(); j++) {
+          var cell = row.getCell(j);
+          if (cell != null) {
+            fn.accept(cell);
+          }
+        }
+      }
+    }
+  }
```

Comment:
I like this pattern.  Excellent way to reuse the loop and make other methods more declarative.

Reply:
Yeah, there were several cases where it was iterating over a range of cells so it made sense to factor that out.